plugin/evm/vm.go

```diff
@@ -620,7 +622,33 @@ func (vm *VM) Shutdown(context.Context) error {
 }
 
 // buildBlock builds a block to be wrapped by ChainState
-func (vm *VM) buildBlock(context.Context) (snowman.Block, error) {
+func (vm *VM) buildBlock(ctx context.Context) (snowman.Block, error) {
+	log.Debug("Trying to build block without context.")
+	return vm.buildBlockWithContext(ctx, nil)
+}
+
+func (vm *VM) buildBlockWithContext(ctx context.Context, proposerVMBlockCtx *block.Context) (snowman.Block, error) {
+	if proposerVMBlockCtx != nil {
+		log.Debug("Building block with context", "pChainBlockHeight", proposerVMBlockCtx.PChainHeight)
+	} else {
+		log.Debug("Attempting to build block with nil context")
```

Comment:
nit: could we make the strings for the 2 debug logs more similar?

Reply:
This is a duplicate log since this is called from `buildBlock`. Can we remove the log in `buildBlock` and only keep the logs in this function?

```suggestion
		log.Debug("Building block without context")
```